### PR TITLE
fix(adapters): WS5 source-mismatch bundle — 7 placeholder + leak fixes

### LINKED ISSUE #1547: ABQ H3 — GOOGLE_CALENDAR extracting date-range sub-heading as haresText for multi-day campout event
## Summary

ABQ H3 "DoT Birthday Weekend Campout" (Fri May 22, 2026) has haresText = "Friday 5/22-Monday 5/25" — a date range, not a hare name.

**HashTracks event:** https://www.hashtracks.xyz/hareline/cmp9uibb6002a04ksb0ct4dbp

**Source:** ABQ H3 Google Calendar (j19gg5vekabk94i8sn3pe892gk@group.calendar.google.com)

**Field affected:** haresText

**Current extracted value:** "Friday 5/22-Monday 5/25"

**Expected value:** null (this is a date range, not a hare name)

**Description in HashTracks:** "RSVP if you plan to camp at DoTs. That way he knows how much to expect you wankers to eat and drink."

## Suspected Adapter

GOOGLE_CALENDAR — ABQ H3 (kennelCode: abqh3)

## Fix Hypothesis

The GOOGLE_CALENDAR adapter for ABQ H3 parses the first line of the GCal event description as haresText. For this multi-day campout, the kennel put the date-range "Friday 5/22-Monday 5/25" as the first description line. The adapter should validate that extracted haresText is not a date-range string before storing it. Prior issue #537 (closed Apr 8, 2026) addressed under-extraction of hares; this is the inverse.

### LINKED ISSUE #1548: Sydney Thirsty H3 — HTML_SCRAPER includes "at the map location below" boilerplate in locationName (source data-entry; 2 affected events)
## Summary

Sydney Thirsty H3 event location fields consistently include the boilerplate phrase "at the map location below" appended to the venue name. This phrase is an editorial instruction referencing an embedded map on the source webpage (sth3.org/upcoming-runs) and is not part of the geocodable address. It degrades map-pin accuracy and clutters the location display.

## Affected Events

| Run | HashTracks | locationName (current) | Expected (clean venue) |
|-----|------------|----------------------|------------------------|
| #1851 | https://www.hashtracks.xyz/hareline/cmoynkypj001504k4twu4qpmk | "Petersham Park, at the map location below" | "Petersham Park" |
| #1852 | https://www.hashtracks.xyz/hareline/cmoynkyqg001704k42jjurn8q | "St Leonards at the map location below" | "St Leonards" |

## Source Evidence (sth3.org/upcoming-runs — verbatim)

```
Thursday May 21st at 6:30pm
Run #1851
Location: Petersham Park, at the map location below
Map: here

Thursday May 28th at 6:30pm
Run #1852
Location: St Leonards at the map location below.
Map: here
```

This is a **source data-entry issue**: the kennel writes "at the map location below" as a literal part of the Location field on their website, referring to a Google Map embed that follows. The HTML_SCRAPER faithfully extracts the string, but the phrase is not geocodable and will cause the map pin to drift or fail.

**Source URL:** https://www.sth3.org/upcoming-runs

## Suspected Adapter

HTML_SCRAPER — Sydney Thirsty H3 Upcoming Runs (sth3.org/upcoming-runs)

## Fix Hypothesis

The scraper should strip the boilerplate suffix ", at the map location below" (and its variants: "at the map location below.", "at the map location below") from the extracted locationName. Adding `"at the map location below"` to the `BRISTOL_LOCATION_OMIT`-style strip list for this source would cleanly resolve both affected events and any future runs that use the same phrasing.

### LINKED ISSUE #1549: RH3 (Richmond VA) — title includes kennel-specific hare-wanted CTA "CLAIM THIS TRAIL" not caught by title-cta-text rule
## Summary

Richmond H3 (RVA) Trail #1695 has title "Richmond H3 Trail #1695: CLAIM THIS TRAIL". The "CLAIM THIS TRAIL" suffix is a kennel-specific hare-wanted CTA pattern — the kennel uses it as a placeholder when no hare has been assigned. The automated `title-cta-text` rule does not flag this because it does not match the current CTA patterns ("wanna hare", "available dates", "check out our", "sign up") nor the CTA_EMBEDDED_PATTERNS set (which covers "volunteer", "hare needed", etc.).

## Evidence

**HashTracks event:** https://www.hashtracks.xyz/hareline/cmmsf82p600at04jtwbrm1hye

**Source:** Richmond H3 Google Calendar (kennelCode: rvah3)

**Field affected:** title

**Current extracted value (verbatim):** "Richmond H3 Trail #1695: CLAIM THIS TRAIL"

**Expected behavior:** The "CLAIM THIS TRAIL" suffix should be recognized as a hare-CTA placeholder and either stripped from the title or flagged as `title-cta-text`.

**Context:** Trail #1695 is on Sun May 24, 2026 (3 days out). The next event (#1696) has a real title "Richmond H3 Trail #1696: Junkyard and Bananta", confirming this is a temporary placeholder, not the intended title.

## Suspected Adapter

GOOGLE_CALENDAR — Richmond H3 Google Calendar (kennelCode: rvah3)

## Fix Hypothesis

Add `"CLAIM THIS TRAIL"` (and potentially `"CLAIM.THIS.TRAIL"` variations) to the `CTA_EMBEDDED_PATTERNS` list in `src/adapters/utils.ts`, or extend the `TITLE_CTA_PATTERN` regex in `src/pipeline/audit-checks.ts` to cover `\bCLAIM\s+THIS\s+TRAIL\b`. This would allow the daily cron to catch future occurrences automatically.

### LINKED ISSUE #1550: BH4 — HTML_SCRAPER extracts haresText = "Open" for run #2002 but source shows hare "Perpencockular"; event subtitle also lost
## Summary

BH4 run #2002 (Sat May 23, 2026) shows haresText = "Open" in HashTracks, but the source page (big-hump.com/hareline.php) clearly shows the hare is "Perpencockular". The event subtitle "Asks For Help To Move" is also missing from the extracted title. This is a source-comparison extraction error.

## Evidence

**HashTracks event:** https://www.hashtracks.xyz/hareline/cmnffx15i000o04js8t5d4ah0

**Source URL:** http://www.big-hump.com/hareline.php

**Field affected:** haresText, title (subtitle missing)

**Current extracted haresText:** "Open"

**Expected haresText (verbatim from source):** "Perpencockular"

**Current extracted title:** "BH4 #2002 @ Brentwood"

**Expected title (verbatim context from source):** "Perpencockular Asks For Help To Move @ Brentwood"

## Source Content (verbatim)

```
Saturday 05/23/2026 #2002
Perpencockular Asks For Help To Move @ Brentwood
```

Compare with a typical run that correctly shows "Open":

```
Wednesday 07/22/2026 #2014
Open @ ???
```

## Suspected Adapter

HTML_SCRAPER — Big Hump H3 Hareline (big-hump.com/hareline.php) — kennelCode: bh4

## Fix Hypothesis

The big-hump.com hareline page uses a two-line format for some events (date/number on line 1, "HARE description @ LOCATION" on line 2) and a single-line format for others. Run #2002 has an event subtitle "Asks For Help To Move" between the hare name and the "@" separator, making the full string "Perpencockular Asks For Help To Move @ Brentwood". The adapter appears to be incorrectly identifying "Perpencockular Asks For Help To Move" as not a hare name, and falling back to a default or nearby "Open" value. The fix should extract everything before " @ " as the title/description, and then apply a heuristic to identify the first word or phrase as the hare name (up to a known CTA pattern like "Asks For Help", "Birthday", etc.) or just store the whole pre-"@" string as the title.

### LINKED ISSUE #1551: Wasatch H3 — GOOGLE_CALENDAR haresText includes event description after hare name (run #1147; "Nipples. A Wasarch/Bash crossover event...")
## Summary

Wasatch H3 run #1147 (Sat May 30, 2026) has haresText = "Nipples. A Wasarch/Bash crossover event. Bring your bike and ride the trails around Bingham Creek Regional park." — the adapter extracted the hare name "Nipples" plus the entire event description sentence as haresText. The hare is "Nipples"; the rest of the text belongs in the description field.

## Evidence

**HashTracks event:** https://www.hashtracks.xyz/hareline/cmnwb4wg2005y04jxgm849k2y

**Source:** Wasatch H3 Google Calendar (shared SLC calendar)

**Field affected:** haresText

**Current extracted haresText (verbatim):** "Nipples. A Wasarch/Bash crossover event. Bring your bike and ride the trails around Bingham Creek Regional park."

**Expected haresText:** "Nipples"

**Description field in HashTracks:** "hare: Nipples. A Wasarch/Bash crossover event. Bring your bike and ride the trails around Bingham Creek Regional park. If you don't ride, I will bring discs for disc golf..."

## Source Context

The GCal event description is:
```
hare: Nipples. A Wasarch/Bash crossover event. Bring your bike and ride the trails around Bingham Creek Regional park.
If you don't ride, I will bring discs for disc golf, and we will wander along the course.

Wine and cheese party at 2:30 at Hammer and Scab's place. Bring wine and/or cheese.

https://www.facebook.com/events/1338205334639619/
```

The kennel uses the format "hare: NAME. event description..." in one continuous sentence on the first line. The adapter's hares extractor matches `hare:` and then captures everything to end-of-line, resulting in "Nipples. A Wasarch/Bash crossover event..." instead of stopping at the period after the hare name.

## Suspected Adapter

GOOGLE_CALENDAR — shared SLC calendar (kennelCode: wasatch-h3)

## Fix Hypothesis

The hares extractor for this GCal source should stop at the first period (or newline) after the hare name, not capture the entire rest of the line. A regex like `/^hare[s]?:\s*([^.\n]+)/i` would correctly capture "Nipples" and leave the description text in the description field.

### LINKED ISSUE #1552: Little Rock H3 — schedule chip renders dangling fragment 'Sundays' with no time (3rd item from 2 configured sources) schedule-label-leak
## Summary

The Little Rock H3 kennel profile chip on HashTracks reads:

```
Wednesdays at 7:00 PM / Sundays at 3:00 PM / Sundays
```

The trailing `/ Sundays` is a dangling third fragment with no time. The audit task setup lists only **2** STATIC_SCHEDULE sources for this kennel:

- Little Rock H3 Static Schedule (Sunday)
- Little Rock H3 Static Schedule (Wednesday)

…yet the chip joins three things. One of two things is happening:

1. There's a third stale/duplicate STATIC_SCHEDULE row in the DB (likely a leftover "Sundays" entry with an empty time field) that's still being concatenated into the schedule chip — but isn't shown in the audit setup because the deep-dive prompt enumerates only "active" sources.
2. The chip-formatter has a bug where it appends a day-only fragment when a source has a populated day but null/empty time.

The HashTracks event list itself (synthesized Sun + Wed events) looks fine — every event card pairs a day with a time (Sun 3:00 PM, Wed 7:00 PM). So the underlying data drives correct events but a junk schedule label.

## Recommended check

Query the `Source` table for `kennelId = lrh3` and look for any extra row beyond the documented two — particularly one with day-of-week = Sunday and time null/empty. If one exists, soft-delete it. If only the two documented sources exist, this is a chip-formatter bug: it should not emit a fragment for a source whose time is empty.

## HashTracks page reference

https://www.hashtracks.xyz/kennels/little-rock-h3 (chip immediately below kennel name)

## Verbatim chip text

`Wednesdays at 7:00 PM / Sundays at 3:00 PM / Sundays`

(Note the inconsistent spacing — `/ Sundays` with no preceding time but the same ` / ` separator as the other two items.)

## Adjacent observation: Sunday time selection

Low priority — not a finding by itself, but: HashTracks renders all synthesized Sunday events at exactly **3:00 PM**, while the kennel's own contact page (`https://lrhash.com/contact`) says Sunday "run time varies from 2-5 p.m." 3:00 PM is a reasonable midpoint, and STATIC_SCHEDULE has no way to express "varies", so I'm not flagging this as a bug — but mentioning here in case the chip rewrite considers a "~2–5 PM" or "3:00 PM (varies)" rendering.

### LINKED ISSUE #1557: Memphis H3 — 2 events have title ending in trailing ' -' (incomplete suffix) title-suffix-leak
## Summary

2 events on `mh3-tn` have titles ending with a trailing ' -' (space-dash):

| Event | Date | aria-label |
|---|---|---|
| `cmoxcei6p002g04l7gcuf2xna` | Mon May 18 2026 | `Memphis H3, GyNO H3 Trail 8 -, Mon, May 18, 6:00 PM` |
| (Apr 12 past event card) | Sat Apr 12 2026 | `Memphis H3, MH3 Trail -, Sat, Apr 12, 2:00 PM` |

## Likely root cause

The title looks like it was built from a template like `{eventName} - {suffix}` where `{suffix}` is empty/missing, leaving a dangling ` -`. The two affected events both use this two-part form:

- `GyNO H3 Trail 8 -` — likely `GyNO H3 Trail {n} - {theme}` where theme is empty
- `MH3 Trail -` — likely `MH3 Trail {n} - {theme}` where both n and theme are empty (the run number being absent suggests this trail wasn't yet numbered on Facebook when the event was created)

Compare against fully-populated examples that work correctly:

- `MH3 Trail 1376 - White Trash Hash: Memphis Unsafe Task Force, a Giggletaint production` (current upcoming, May 23)
- `Buffy & Mission A's Birthday Trail (MH3 Trail 1375)` (past, May 9)

Fix: trim trailing ` -` (and any other terminal separator) when concatenating template segments, or skip the separator entirely when the suffix is empty.

## Adapter scope

Both sources for this kennel are candidates:

- `GOOGLE_CALENDAR` (memphish3@gmail.com) — the underlying calendar shows `MH3 Trail` as the raw event title; the ` -` suffix might be appended by our enrichment step (cross-referencing the FB hosted events for run numbers + themes).
- `FACEBOOK_HOSTED_EVENTS` (`facebook.com/MemphisH3/upcoming_hosted_events`) — could also be the originating template.

Not speculating further without code; tagging the maintainers of `audit-issue-sync` / calendar-event-title-builder.

## Cross-references

- Same root-cause shape as #756 (Moooouston trailing-dash, 1 issue covered all events) and #1060 (Space City trailing-colon). Bundling these two events into one finding per the root-cause-bundle rule.
- Issue #1556 (filed in parallel) flags the GyNO event as a cross-kennel-conflation independent of the title artifact — both issues touch the same event but the root causes are distinct.

### PR DESCRIPTION
**Pull request by johnrclem**
## Summary

Quick Win bundle of 7 disjoint source-mismatch fixes — 4 GCal-adapter
fixes, 1 FB-parser fix, 1 HTML-scraper fix, 1 HTML-scraper data-entry
boilerplate, and 1 UI schedule-chip dedup.

## Fixes

| Issue | Where | Fix |
|---|---|---|
| #1547 | `src/adapters/google-calendar/adapter.ts` + `hare-extraction.ts` | Parenthetical date-range `(Friday 5/22-Monday 5/25)` treated as instructional → stripped from title, not extracted as hares. Defense-in-depth `DATE_RANGE_RE` in `cleanAndFilterHares`. |
| #1549 | `src/adapters/utils.ts` | `CLAIM THIS TRAIL` added to `CTA_EMBEDDED_PATTERNS` (GCal skips event + audit-checks flags `title-cta-text`). |
| #1551 | `src/adapters/hare-extraction.ts` | `cleanAndFilterHares` truncates at first `. ` when the tail is sentence-shaped (3+ tokens AND contains a lowercase token). `Dr. Strange` and `Mr. Happy. Big Fun Bob` survive (Codex review). |
| #1557 | `src/adapters/facebook-hosted-events/parser.ts` | FB title strips trailing `[-–—:]` delimiter (mirrors GCal #756 / #1060). |
| #1548 | `src/adapters/html-scraper/sydney-thirsty-h3.ts` | `MAP_BOILERPLATE_RE` strips `at the map location below` from `Location:` field. |
| #1550 | `src/adapters/html-scraper/big-hump.ts` | (a) `parseHaresFromDescription` rejects `"Open"`. (b) Rule 5b matches `<hare> <action-verb> …` with lazy capture (`Captain Hook Says Goodbye` → `Captain Hook`). (c) Resolved-hares boundary clears `"Open"` only when venue is also unknown (real hasher named `Open` at a known venue survives). (d) `h4` titles with descriptive subtitles preserved instead of being rewritten to `BH4 #N @ Venue`. |
| #1552 | `src/lib/format.ts` | `formatScheduleRules` drops bare-day fragments subsumed by a peer with same day + time (`Sundays` dropped when `Sundays at 3:00 PM` peer exists). |

## Live verification

`scripts/live-verify-ws5-bundle.ts` runs all 6 adapters against production
sources (`ABQ H3`, `Richmond H3`, `Whoreman H3 Calendar` for Wasatch,
`Memphis H3 FB`, `Sydney Thirsty H3`, `Big Hump H3`). All checks clean:

```
## ABQ H3 Google Calendar — events: 94 — haresText with M/D date tokens: 0 (expect 0)
## Richmond H3 Google Calendar — events: 2 — titles containing 'CLAIM THIS TRAIL': 0 (expect 0)
## Whoreman H3 Calendar — events: 147 — haresText with sentence trailer: 0 (expect 0)
## Memphis H3 Facebook Hosted Events — events: 8 — titles ending in trailing delimiter: 0 (expect 0)
## Sydney Thirsty H3 Upcoming Runs — events: 3 — location containing 'at the map location below': 0 (expect 0)
## Big Hump H3 Hareline — events: 102 — haresText literal 'Open': 0 (expect 0)
✓ All targets clean
```

## Test plan

- [x] `npx tsc --noEmit` — clean
- [x] `npm run lint` — 0 errors (20 pre-existing warnings)
- [x] `npm test` — 6970 passed, 0 failed
- [x] Live verify against 6 production sources (above)
- [x] `/codex:adversarial-review` — 3 findings addressed:
  - High: action-verb rule now matches multi-word hares
  - Medium: `Open` filter gated on `venueUnknown`
  - Medium: sentence-trailer requires lowercase-token in tail
- [x] `/simplify` — comment trimming + chain reductions applied

## Sequencing note

WS1 (issue #1233 Chicagoland `kennelPatterns`) wasn't open at branch
time; WS5 was branched from `main`. The conflict surface is the GCal
adapter, but WS5's edits are in title/hares extraction (separate code
paths from WS1's matcher logic) so a merge should be clean.

Closes #1547
Closes #1548
Closes #1549
Closes #1550
Closes #1551
Closes #1552
Closes #1557

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by johnrclem** [comment]
@claude @codex review

File: src/adapters/html-scraper/big-hump.ts [hunk 1]
```diff
@@ -489,10 +511,21 @@ export class BigHumpAdapter implements SourceAdapter {
...
+          // if it equals `hares` (case-insensitive), there's no subtitle.
           let title = titleFromH4;
           if (runNumber) {
-            title = location ? `BH4 #${runNumber} @ ${location}` : `BH4 #${runNumber}`;
+            const atIdx = h4Text.lastIndexOf(" @ ");
+            const h4HarePart = (atIdx >= 0 ? h4Text.slice(0, atIdx) : h4Text).trim();
+            const isBareHareTitle =
+              !hares || h4HarePart.toLowerCase() === hares.toLowerCase();
+            if (isBareHareTitle) {
+              title = location ? `BH4 #${runNumber} @ ${location}` : `BH4 #${runNumber}`;
```
[the remark this answers is not in the document]
Comment (johnrclem) [reply src/adapters/html-scraper/big-hump.ts]: Fixed in 7a2d88c6 — inverted the logic to require `hares` truthy AND match harePart before rewriting, and hoisted into a `buildBh4Title` helper so unparsed h4 titles like "Bungle in the Jungle @ Steelville" survive. New regression test in big-hump.test.ts.

File: src/adapters/hare-extraction.ts [hunk 2]
```diff
@@ -60,6 +60,22 @@ const COHARE_COMMENTARY_RE = /\s*(?:could|need)\s+.*?co-?hares?\b.*$/i; // NOSON
...
+// Description-sentence leak (#1551 Wasatch): kennel uses "hare: NAME. event
+// description..." on one line. The greedy `(.*)` capture pulls the entire
+// remainder. Truncate at the first ". " when the post-period tail is sentence-
+// shaped — 3+ tokens AND at least one token starts with a lowercase letter.
+// Hash names use Title Case; sentence tails contain lowercase function words
+// like "crossover", "is", "and". This refinement preserves period-prefixed
+// names like "Dr. Strange. Captain Hook" (tail all Title Case → no truncate)
+// while still catching "Nipples. A Wasarch/Bash crossover event…" (Codex
+// review on this PR).
+const HARE_SENTENCE_BOUNDARY_RE = /\.\s+/;
```
[the remark this answers is not in the document]
Comment (johnrclem) [reply src/adapters/hare-extraction.ts]: Fixed in 7a2d88c6 — went with a per-position scan (`findHareSentenceBoundary`) over the lookbehind approach since `\b` inside lookbehind has engine-specific behavior. The scan now skips any `. ` preceded by a common honorific (Dr/Mr/Ms/Mrs/St) and continues to the next boundary. 3 new tests cover the edge case.

File: src/adapters/html-scraper/big-hump.ts [hunk 3]
```diff
@@ -211,6 +221,9 @@ function parseHaresFromDescription(text: string): string | undefined {
   const name = match[1].trim();
   // "away: …" is departure time, not a hare name
   if (/^away/i.test(name)) return undefined;
+  // #1550: "Open" is a "hare slot available" placeholder, not a real name.
+  // Mirrors the upstream h4 "Open @ ???" guard in the adapter fetch().
+  if (/^open$/i.test(name)) return undefined;
```
[the remark this answers is not in the document]
Comment (johnrclem) [reply src/adapters/html-scraper/big-hump.ts]: Fixed in 68998e45 — removed the unconditional filter in `parseHaresFromDescription` and added `pickHaresPreferringRealName` so descHares='Open' yields to titleHares only when titleHares is a non-Open value. The final venueUnknown gate at the convergence point still clears 'Open' for unknown venues. New regression test in big-hump.test.ts.

File: src/adapters/hare-extraction.ts [hunk 4]
```diff
@@ -156,8 +172,25 @@ function cleanAndFilterHares(raw: string): string | undefined {
...
+  // ". " when the tail is sentence-shaped (3+ tokens, at least one starting
+  // with a lowercase letter). Preserves period-prefixed names like
+  // "Dr. Strange" and even "Dr. Strange. Captain Hook" — tails of pure
+  // Title-Case tokens are kept.
+  const periodIdx = hares.search(HARE_SENTENCE_BOUNDARY_RE);
+  if (periodIdx >= 0) {
+    const tail = hares.slice(periodIdx).replace(HARE_SENTENCE_BOUNDARY_RE, "");
+    const tailTokens = tail.split(/\s+/).filter(Boolean);
+    if (tailTokens.length >= 3 && LOWERCASE_TOKEN_RE.test(tail)) {
+      hares = hares.slice(0, periodIdx).trim();
```
[the remark this answers is not in the document]
Comment (johnrclem) [reply src/adapters/hare-extraction.ts]: Addressed in 7a2d88c6 — `findHareSentenceBoundary` now scans each `. ` position and skips ones preceded by a honorific (Dr/Mr/Ms/Mrs/St), so `Hare: Dr. Evil and Mini Me` no longer truncates at `Dr`. 3 new tests cover the edge case.
